File: dotnet/DesignPatterns/BehavioralDesignPatterns/IteratorDesignPattern/IteratorDesignPattern.Example3/Song.cs
namespace IteratorDesignPattern.Example3;

public class Song
{
    public string Id { get; set; } = string.Empty;
    public string Title { get; set; } = string.Empty;
    public string Artist { get; set; } = string.Empty;
    public string Album { get; set; } = string.Empty;
    public TimeSpan Duration { get; set; }
    public string Genre { get; set; } = string.Empty;

    public override string ToString()
    {
        return $"{Title} - {Artist} ({Duration:mm\\:ss})";
    }
}

File: nodejs/DesignPatterns/BehavioralDesignPatterns/IteratorDesignPattern/IteratorDesignPattern.Example3/src/Song.ts
export class Song {
  constructor(
    public readonly id: string,
    public readonly title: string,
    public readonly artist: string,
    public readonly album: string,
    public readonly durationSeconds: number,
    public readonly genre: string
  ) {}

  toString(): string {
    const min = Math.floor(this.durationSeconds / 60)
      .toString()
      .padStart(2, "0");
    const sec = (this.durationSeconds % 60).toString().padStart(2, "0");
    return `${this.title} - ${this.artist} (${min}:${sec})`;
  }
}
